wallet/wallet.c

```diff
@@ -377,6 +378,43 @@ const struct utxo **wallet_select_coins(const tal_t *ctx, struct wallet *w,
 	return utxo;
 }
 
+const struct utxo **wallet_select_specific(const tal_t *ctx, struct wallet *w,
+					const char **outpoints)
+{
+	size_t i, j;
+	char **outpoint;
+	unsigned int output_index;
+	struct bitcoin_txid *txid;
+	struct utxo **available = wallet_get_utxos(ctx, w, output_state_available);
```

Comment:
available should either be allocated off NULL (which by convention, indicates you're explicitly going to take/free it, as you do below), or just off tmpctx and forget about it.